bindings.go

```diff
@@ -17,63 +17,141 @@ const (
 // declared and bound. If the ExchangeName is not defined (an empty string), the
 // queue will not be bound to the exchange but assumed to use the default match.
 type HandlerBinding struct {
-	QueueName    string
+	// QueueName is the name of the queue that the handler should be bound to.
+	QueueName string
+
+	// ExchangeName is the exchange that the queue should be bound to.
 	ExchangeName string
-	ExchangeType string
-	RoutingKey   string
-	BindHeaders  amqp.Table
-	Handler      HandlerFunc
+
+	// RoutingKey is the routing key that the queue should be bound to.
+	RoutingKey string
+
+	// BindHeaders is the headers that the queue should be bound to.
+	BindHeaders amqp.Table
+
+	// Handler is the function that should be called when a message is
+	// received.
+	Handler HandlerFunc
+
+	// QueueDurable sets the durable flag. Should not be used together with
+	// QueueExclusive since that will fail in a future RabbitMQ version.
+	QueueDurable bool
+
+	// Set the queue to be automatically deleted when the last consumer stops.
+	QueueAutoDelete bool
+
+	// SkipQueueDeclare will skip the queue declaration. You can be useful when
+	// you are migrating queue types or arguments and want to avoid the
+	// PRECONDITION_FAILED error. If this is set to true, you cannot use auto
+	// generated queue names.
+	SkipQueueDeclare bool
+
+	// Exclusive queues can only be used by the connection that created them.
+	// If this is true, the queue will be deleted during a network failure
+	// making any requests fail before the reconnect can happen.
+	QueueExclusive bool
+
+	// QueueArgs sets any extra queue arguments.
+	QueueArgs amqp.Table
+
+	// AutoAck sets the auto-ack flag on the consumer.
+	AutoAck bool
+
+	// PrefetchCount sets the prefetch count for the consumer. This is only
+	// really usable when AutoAck is also set to false.
+	PrefetchCount int
+
+	// ConsumerArgs sets any extra consumer arguments.
+	ConsumerArgs amqp.Table
 }
 
-// DirectBinding returns a HandlerBinding to use for direct exchanges where each
-// routing key will be mapped to one handler.
-func DirectBinding(routingKey string, handler HandlerFunc) HandlerBinding {
-	return HandlerBinding{
-		QueueName:    routingKey,
-		ExchangeName: "amq.direct",
-		ExchangeType: ExchangeDirect,
-		RoutingKey:   routingKey,
-		BindHeaders:  amqp.Table{},
-		Handler:      handler,
+// With will apply the given functions to the HandlerBinding. This is a
+// convenient way of setting options.
+func (b HandlerBinding) With(funcs ...func(*HandlerBinding)) HandlerBinding {
+	for _, f := range funcs {
+		f(&b)
 	}
+
+	return b
 }
 
-// FanoutBinding returns a HandlerBinding to use for fanout exchanges. These
```

Comment:
Is fanout being deprecated or is it just that it makes less sense in an RPC setup like this?

Reply:
Never mind 2, saw the part in the commit message and actual pull request as well 🙈